kolibri/plugins/setup_wizard/assets/src/views/OnboardingStepBase.vue

```diff
@@ -0,0 +1,234 @@
+<template>
+
+  <!-- Always base-container class, wrapping screen size class when small -->
+  <div
+    :class="{ 'base-container': true, 'windowIsSmall': windowIsSmall }"
+    :style="{ 'background-color': windowIsSmall ? $themeTokens.surface : '' }"
+  >
+
+    <div class="logo-lang-container">
+      <!--
+        Small Screen: Back button (if not hidden by prop)
+        Med-Lg Screen: Logo
+        -->
+      <div v-if="!windowIsSmall" class="logo-wrapper">
+        <CoreLogo class="logo-image" />
+        <span class="logo-text">{{ coreString('kolibriLabel') }}</span>
+      </div>
+
+      <KIconButton
+        v-else-if="!noBackAction"
+        class="back-icon-button"
+        icon="back"
+        @click="$emit('back')"
+      />
+
+      <!-- Language switcher visible regardless of screen size -->
+      <KButton
+        class="languages-button"
+        icon="language"
+        :text="selectedLanguage.lang_name"
+        :primary="false"
+        appearance="flat-button"
+        @click="showLanguageModal = true"
+      />
+    </div>
+
+    <!-- topMargin & noPadding props for KPageContainer - harmless when case of div -->
+    <component
+      :is="windowIsSmall ? 'div' : 'KPageContainer'"
+      :topMargin="16"
+      :noPadding="true"
+    >
+      <div class="content">
+        <slot></slot>
+      </div>
+
+      <!-- Border hidden on mobile by making it the same as the background -->
+      <div
+        class="footer"
+        :style="{
+          borderTop: `1px solid ${windowIsSmall ? $themeTokens.surface : $themeTokens.fineLine}`
+        }"
+      >
+        <!-- No room for slot on small screens -->
+        <div v-if="!windowIsSmall" class="footer-section">
+          <slot name="footer"></slot>
```

Comment:
I'm unclear on if this footer slot is distinct from the KButtonGroup footer below? Is this where the text such as "Import Facility 1 of 4" will go? If so, maybe _slightly_ more description would be helpful